### ISSUE TITLE
Multiple decorator error when using `@schema.output`

### ISSUE THREAD

**Issue by sT0v** [DAGWorks-Inc/hamilton#1163]
This is pretty self-explanatory, but just wanted to make sure that this was in fact illegal?
```@schema.output(
  ("ticker", "str"),
  ("fee", "float")
)
@tag(table_type="reference")
def ref_table_1() -> pd.DataFrame:
  ...```
```ValueError: Got multiple decorators for decorator @<class 'hamilton.function_modifiers.metadata.SchemaOutput'>. Only one allowed.```

[Slack Message](https://hamilton-opensource.slack.com/archives/C03M33QB4M8/p1727897401998989?thread_ts=1727897401.998989&cid=C03M33QB4M8)

**Comment by elijahbenizzy** [comment]
Yeah, this should work... I think there's some bug in how we handle this:
- there's ohnly one schema.output there
- But it's triggering this... https://github.com/DAGWorks-Inc/hamilton/blob/acd2fc75692cfbe4c98eede3f5e89988c3950f1a/hamilton/function_modifiers/metadata.py#L242

What's happening is it's going by lifecycle step, but it should really go by class... The `SchemaOutput` is not respecting the contract (E.G. that you can't override allow_multiple). But we should clarify this, or just allow multiple and have a check on the tag decorator to ensure they don't overwrite.
